upload/system/library/Packetery/Vendor/AddEditPage.php

```diff
@@ -0,0 +1,340 @@
+			];
+		}
+
+		return $internalVendors;
+	}
+
+	/**
+	 * @param array $formData
+	 *
+	 * @return array
+	 */
+	public function getAddVendorFormValidationErrors(array $formData) {
+		$errors = [];
+		$weightRulesErrors = [];
+
+		if ($formData['action'] === ControllerExtensionShippingZasilkovna::ACTION_ADD_VENDOR) {
+			if (empty($formData['vendor'])) {
+				$errors[] = $this->language->get('vendor_add_error_required_vendor');
+			}
+
+			if (isset($formData['weight_rules'])) {
+				$formData['weight_rules'] = $this->removeEmptyWeightRules($formData['weight_rules']);
+			}
+
+			if (empty($formData['weight_rules'])) {
+				$errors[] = $this->language->get('vendor_add_error_weight_rules_missing');
+			} else {
+				$weightRulesErrors = $this->validateWeightRules($formData['weight_rules']);
+			}
+		}
+
+		return array_merge($errors, $weightRulesErrors);
+	}
+
+	/**
+	 * @param array $weightRules
+	 *
+	 * @return array
+	 */
+	private function validateWeightRules(array $weightRules) {
+		$errors = [];
+
+		foreach ($weightRules as $rule) {
+			if (!isset($rule['max_weight']) || !is_numeric($rule['max_weight']) || $rule['max_weight'] <= 0) {
+				$errors[] = 'vendor_add_error_rule_max_weight_invalid';
+			}
+
+			if (!isset($rule['price']) || !is_numeric($rule['price']) || $rule['price'] <= 0) {
+				$errors[] = 'vendor_add_error_rule_price_invalid';
+			}
+		}
+
+		if ($this->checkDuplicateWeightRules($weightRules)) {
+			$errors[] = 'vendor_add_error_rule_duplicate_weights';
+		}
+
+		return $errors;
+	}
+
+	/**
+	 * @param array $weightRules
+	 *
+	 * @return bool
+	 */
+	private function checkDuplicateWeightRules(array $weightRules) {
+		$uniqueWeights = [];
+
+		foreach ($weightRules as $rule) {
+			$maxWeight = $rule['max_weight'];
+
+			if (isset($uniqueWeights[$maxWeight])) {
+				return true; // Duplicate weight found
+			}
+
+			$uniqueWeights[$maxWeight] = true;
+		}
+
+		return false; // No duplicate weights found
+	}
+
+	/**
+	 * Session will contain invalid form data for form pre-filling
+	 *
+	 * @param array $formData
+	 *
+	 * @return void
+	 */
+	private function saveAddVendorFormDataToSession(array $formData) {
+		$this->session->data['vendor_add_form_data'] = $formData;
+	}
+
+	/**
+	 * @return array
+	 */
+	private function getAddVendorFormDataFromSession() {
+		if (!isset($this->session->data['vendor_add_form_data'])) {
+			return [];
+		}
+
+		$formData = $this->session->data['vendor_add_form_data'];
+		unset($this->session->data['vendor_add_form_data']);
+
+		return $formData;
+	}
+
+	/**
+	 * @param array $formData
+	 *
+	 * @return array
+	 */
+	public function getFormDefaults(array $formData = []) {
+
+		if (empty($formData)) {
+			$formData = $this->getAddVendorFormDataFromSession();
+		}
+		$prefilledData = [];
+		foreach ($formData as $key => $value) {
+			if (in_array($key, ['vendor', 'country', 'vendor_group', 'cart_name', 'free_shipping_limit', 'is_enabled', 'weight_rules'])) {
```

Comment:
Toto není dobrý způsob řešení.  Musíš dávat pozor, aby jsi v tomto poli neměl stejný klíč jako v poli $data v metodě add_vendor(), protože tam děláš array_merge($data, $prefilledData).   Druhým důvodem je to, že když v budoucnu rozšíříš formulář o další pole, tak si budeš muset pamatovat,  že musíš jít do této metody a zde upravit obsah pole.  To je velice náchylné na chyby.